tests/vm_startup_test.go

```diff
@@ -0,0 +1,162 @@
+			err = getClient().Delete(context.Background(), &vmiObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			vmiList := &v1.VirtualMachineInstanceList{}
+			err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(vmiList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
+	})
+
+	AfterEach(func() {
+		By("Removing all Network AttachmentDefinitions from the namespace")
+		nadList := &networkv1.NetworkAttachmentDefinitionList{}
+		err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, nadObject := range nadList.Items {
+			err = getClient().Delete(context.Background(), &nadObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			nadList := &networkv1.NetworkAttachmentDefinitionList{}
+			err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(nadList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all nad objects")
+	})
+
+	AfterEach(func() {
+		By("Removing the test namespace")
+		namespace := &k8sv1.Namespace{}
+		err := getClient().Get(context.Background(), types.NamespacedName{Name: testNamespace}, namespace)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			return getClient().Delete(context.Background(), namespace)
+		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s'", testNamespace))
+	})
+
+	Context("with one secondary and no default interface", func() {
+		It("should get the correct IP when nslookup to the secondary interface FQDN", func() {
+			By("Creating a NetworkAttachmentDefinition")
+			err := createNetworkAttachmentDefinition(testNamespace, "ptp-conf")
+			Expect(err).ToNot(HaveOccurred(), "Should successfully create the NAD")
+
+			By("Creating a VirtualMachineInstance")
+			interfaceName := "nic1"
+			interfaces := []v1.Interface{{Name: interfaceName, InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}}}
+			networks := []v1.Network{
+				{Name: interfaceName, NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: "ptp-conf"},
+				}},
+			}
+			vmi := CreateVmiObject("vmi-sec", testNamespace, interfaces, networks)
+			vmiName := vmi.Name
+			err = getClient().Create(context.Background(), vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Invoking nslookup on the VMI interface FQDN")
+			Eventually(func() int {
+				vmi = &v1.VirtualMachineInstance{}
+				err = getClient().Get(context.Background(), types.NamespacedName{Namespace: testNamespace, Name: vmiName}, vmi)
+				return len(vmi.Status.Interfaces)
```

Comment:
the fact you have an interface doesn't mean it has an ip yet theoretically
guest-agent might just advertise it bit later
maybe the ptp is the reason we can trust it here specifically

maybe better to make it generic ?

Reply:
done